lib/Doctrine/DBAL/Driver/OCI8/OCI8Statement.php

```diff
@@ -134,7 +139,15 @@ static public function convertPositionalToNamedPlaceholders($statement)
      */
     public function bindValue($param, $value, $type = null)
     {
-        return $this->bindParam($param, $value, $type, null);
+        if ($type === \PDO::PARAM_LOB || -99 === $type) {
```

Comment:
I assumed this would lead to BC break. I thought about that but the given value will always be a `string`, we still need to choose whether it's supposed to be stored as a CLOB or as a BLOB.

We could however send an oci descriptor to the `bindValue` directly. This would allow the following test to work:

```
public function bindParam($column, &$variable, $type = null, $length = null)
    if (is_a($variable, 'Oci-Lob')) {
        return oci_bind_by_name($this->_sth, $column, $variable, -1, $type === PDO::PARAM_LOB ? : OCI_B_BLOB : OCI_B_CLOB);
    }
    //...
}
```

I may give this a shot, I need to make sure that the `PDO::PARAM_LOB` always correspond to a `BLOB` and not a `CLOB` (not sure about this). Also, I need to find where I could create the oci descriptor if not in the statement.